src/UI/HTMLTooltip.js

```diff
@@ -53,6 +63,22 @@ export class HTMLTooltip {
         // @ts-ignore how to narrow this.host to HTMLElement?
         addInlineStyles(this.host, forcedVisibilityStyles)
         this.count = 0
+        /**
+         * @type {{
+         *   'tooltip': TransformRuleObj,
+         *   'caret': TransformRuleObj
+         * }}
+         */
+        this.transformRules = {
+            caret: {
+                getRuleString: this.options.caretPositionClass,
+                index: null
+            },
+            tooltip: {
+                getRuleString: this.options.tooltipPositionClass,
+                index: null
```

Comment:
This allows us to keep two separate indexes for the rules we add to the stylesheet, so we can delete them appropriately.